From 76e3c3db4cc9b7e4f86d3efeacf17a689d43c74f Mon Sep 17 00:00:00 2001
From: David Edmundson <davidedmundson@kde.org>
Date: Thu, 03 Oct 2024 16:32:30 +0100
Subject: [PATCH] client: Update cached buffer size on startup

EGL Windows cache the buffer size to use, so that we can access it from
multiple threads safely later. This is updated every time application
code is resized including the first configure event.

There may be an existing size at the time the window is created, and
application code could call makeCurrent before the window has the first
expose event. To cover this we should initialize the default
sizeWithMargins on creation. The rest of the updateSurface method will
do nothing as we have no existing window and create is false.

Fixes: QTBUG-129630
Fixes: QTBUG-129403
Pick-to: 6.7
Pick-to: 6.7
Change-Id: I403a5f834bb5da5c34e712927c652033aa9bca3e
Reviewed-by: Vlad Zahorodnii <vlad.zahorodnii@kde.org>
(cherry picked from commit ae741501eed46e6210e5967e3473e8ffc489f224)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---

diff --git a/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp b/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp
index 8be0b47..bf056e4 100644
--- a/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp
+++ b/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp
@@ -28,6 +28,7 @@
         m_clientBufferIntegration = static_cast<QWaylandEglClientBufferIntegration *>(
                 mDisplay->clientBufferIntegration());
     });
+    ensureSize();
 }
 
 QWaylandEglWindow::~QWaylandEglWindow()
